crates/compiler/builtins/bitcode/src/str.zig

```diff
@@ -1306,6 +1306,43 @@ test "countGraphemeClusters: emojis, ut8, and ascii characters" {
     try expectEqual(count, 10);
 }
 
+// Str.graphemes
+pub fn strGraphemes(string: RocStr) callconv(.C) RocList {
+    var list = RocList.allocate(@alignOf(RocStr), countGraphemeClusters(string), @sizeOf(RocStr));
+    const graphemes = @ptrCast([*]RocStr, @alignCast(@alignOf(RocStr), list.bytes));
+
+    const bytes_ptr = string.asU8ptr();
+    var bytes = bytes_ptr[0..string.len()];
+    var iter = (unicode.Utf8View.init(bytes) catch unreachable).iterator();
+    var grapheme_break_state: ?grapheme.BoundClass = null;
+    var grapheme_break_state_ptr = &grapheme_break_state;
+    var opt_last_codepoint: ?u21 = null;
+
+    var list_index: usize = 0;
+    var start_index: usize = 0;
+    var str_index: usize = 0;
+    var cur_codepoint_len: usize = 0;
+
+    while (iter.nextCodepoint()) |cur_codepoint| {
+        cur_codepoint_len = unicode.utf8CodepointSequenceLength(cur_codepoint) catch unreachable;
+        if (opt_last_codepoint) |last_codepoint| {
+            var did_break = grapheme.isGraphemeBreak(last_codepoint, cur_codepoint, grapheme_break_state_ptr);
+            if (did_break) {
+                graphemes[list_index] = RocStr.init(bytes_ptr + start_index, str_index - start_index + cur_codepoint_len);
+                list_index += 1;
+                start_index = str_index + cur_codepoint_len;
+                grapheme_break_state = null;
```

Comment:
This variable only ever seems to be null. Are you sure it's needed? Or could there be a bug where something is supposed to be using it and isn't?

Reply:
The `isGraphemeBreak()` function seems to use it to determine the state, I'm not too sure about what or why it is being used

https://github.com/roc-lang/roc/blob/da0a8f0b2b7ceae63adb2dec1070269ae53cbae6/crates/compiler/builtins/bitcode/src/helpers/grapheme.zig#L146-L150

I think we can safely get rid of it and use `null` in the function call instead

Reply:
I just saw the docs on Unicode Text Segmentation (https://unicode.org/reports/tr29/) and it looks like we need to be setting `grapheme_break_state` to 0 rather than null. I tried it, and while it does partially fix the issue, it introduces a new one where the emoji strings in the RocList are mangled with � everywhere